controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -276,6 +276,9 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return ctrl.Result{}, nil
 	}
 
+	// Set LocalAPIServerPort from the cluster spec
+	r.setLocalAPIServerPort(cluster, kcp)
```

Comment:
Q: Have you considered a slightly different version where this default happens in CABPK
nit.
I'm asking because in CABPK we already have `reconcileTopLevelObjectSettings` that does a similar job (it applies defaults into ClusterConfiguration starting from Cluster settings); If we can add `reconcileTopLevelMachineSettings` with this PR then all the defaulting logic will be in one sigle place/controller

Reply:
not against moving it there, updating